letsencrypt/renewer.py

```diff
@@ -166,6 +167,16 @@ def main(config=None, cli_args=sys.argv[1:]):
     # specify a config file on the command line, which, if provided, should
     # take precedence over this one.
     config.merge(configobj.ConfigObj(cli_config.renewer_config_file))
+    # Ensure that all of the needed folders have been created before continuing
+    uid = os.geteuid()
+    if (not os.path.isdir(cli_config.renewal_configs_dir) or
+        not os.path.isdir(cli_config.config_dir)):
+        print "Could not find config directory. Exiting. "
```

Comment:
Personally I'd just remove the check and print statement and let it fail on it's own. In my opinion, this is already a pretty descriptive error:

```
OSError: [Errno 2] No such file or directory: '/tmp/le/conf/renewal'
```

Additionally, to do the check properly, we'd also have to check for permissions, but we get a pretty nice error there already as well:

```
OSError: [Errno 13] Permission denied: '/tmp/tmp.L93C74Yudw/renewal'
```

If someone really would prefer to raise one of our custom exception types, I think the better approach would be to just wrap the code that may throw the two exceptions above in a try block, catch them, and handle them appropriately. I don't think this is necessary though and certainly isn't needed to merge this PR.